### PR TITLE
State management for Data Visualizer

### PR DESCRIPTION
**Pull request by ajpal**
# Description
- clear all state when `this.props.tableName` changes, using React key
- clear selected columns when chart type changes
- clear filter value when filter column changes

## Testing story

# Reviewer Checklist:

- [x] Tests provide adequate coverage
- [x] Code is well-commented
- [x] New features are translatable or updates will not break translations
- [x] Relevant documentation has been added or updated
- [x] User impact is well-understood and desirable
- [x] Pull Request is labeled appropriately
- [x] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/storage/dataBrowser/TableControls.jsx [hunk 1]
```diff
@@ -67,7 +67,7 @@ class TableControls extends React.Component {
         </div>{' '}
         <div style={styles.buttonWrapper}>
           {experiments.isEnabled(experiments.APPLAB_DATASETS) && (
-            <VisualizerModal />
+            <VisualizerModal key={this.props.tableName} />
```
Comment (islemaster) [inline apps/src/storage/dataBrowser/TableControls.jsx:70]: Interesting!  I hadn't seen `key` used this way before, so I did some digging and learned that this is the ["Fully uncontrolled component with a `key`"](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key) pattern, recommended [by the React docs](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops) when "you want to 'reset' some state when a prop changes."  Great find!

**Review by islemaster — CHANGES_REQUESTED** [review]
There are three new behaviors in this PR.  Please cover them with new unit tests.  Otherwise, this looks great!  It feels like you've built a component that's easy to change. :+1:

File: apps/test/unit/storage/dataBrowser/TableControlsTest.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      wrapper
+        .find(VisualizerModal)
+        .children()
+        .first()
+        .state().chartType
+    ).to.equal(ChartType.BAR_CHART);
+    wrapper.setProps({
+      children: React.cloneElement(wrapper.props().children, {
+        tableName: 'differentTable'
+      })
```
Comment (islemaster) [inline apps/test/unit/storage/dataBrowser/TableControlsTest.js]: Wow!  Looks like there were some tricky problems to solve writing this test.  Nice work.

One thing I like to explore when actions and assertions start getting verbose like this is extracting helper methods just to make the test easier to read. I've added a couple of commits to do that, and make the assertions align a little better with the idea of the state being fully reset.  Let me know what you think.

**Review by islemaster — APPROVED** [review]
Great work!

**Review by epeach — APPROVED** [review]
LGTM

**Pull request merged**